lib.nix

```diff
@@ -12,10 +12,32 @@ let
        then result
        else default;
 
+  # Removes files within a Haskell source tree which won't change the
+  # result of building the package.
+  # This is so that cached build products can be used whenever possible.
+  # It also applies the lib.cleanSource filter from nixpkgs which
+  # removes VCS directories, emacs backup files, etc.
+  cleanSourceTree = src:
```

Comment:
Can we add doc files to this function as well so changing md files doesn't trigger new builds?

Reply:
Probably not because there is usually `extra-source-files: README.md` in the cabal file.